src/Internal/FulfilledPromise.php

```diff
@@ -56,9 +56,14 @@ public function catch(callable $onRejected): PromiseInterface
         return $this;
     }
 
+    /**
+     * @param callable(): (void|PromiseInterface<void>) $onFulfilledOrRejected
+     * @return PromiseInterface<T>
+     */
```

Comment:
Did this and `src/Promise.php` become an issue in PHP 8.4 + the PHPStan upgrade? What where the errors?